etl/stores/arkg_store.py

```diff
@@ -0,0 +1,88 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Self
+
+import pyoxigraph as ox
+
+from etl.models import AntiRecommendationGraphTuple
+from etl.models.types import RdfMimeType
+from etl.pipelines.arkg_builder_pipeline import ArkgBuilderPipeline
+
+
+class ArkgStore:
+    """A database to store an Anti-Recommendation Knowledge Graph."""
+
+    @dataclass(frozen=True)
+    class Descriptor:
+        """A dataclass that holds the Path of the resource stored in database."""
+
+        arkg_store_path: Path
+
+    def __init__(self, *, arkg_store: ox.Store, arkg_store_path: Path) -> None:
```

Comment:
Nit: would call this `store` and `directory_path` respectively. Same with private variables. You don't need to prefix them when it's clear from context.